Cython/Includes/cpython/datetime.pxd

```diff
@@ -146,25 +195,50 @@ cdef inline void import_datetime():
+
+# Create a datetime object with a fold using DateTime CAPI factory function.
+cdef inline object datetime_and_fold_new(int year, int month, int day, int hour, int minute, int second, int microsecond, object tz, int fold):
+    return PyDateTimeAPI.DateTime_FromDateAndTimeAndFold(year, month, day, hour, minute, second, microsecond, tz, fold, PyDateTimeAPI.DateTimeType)
+
 # Create time object using DateTime CAPI factory function
 # Note, there are no range checks for any of the arguments.
 cdef inline object time_new(int hour, int minute, int second, int microsecond, object tz):
     return PyDateTimeAPI.Time_FromTime(hour, minute, second, microsecond, tz, PyDateTimeAPI.TimeType)
 
-# Create datetime object using DateTime CAPI factory function.
-# Note, there are no range checks for any of the arguments.
-cdef inline object datetime_new(int year, int month, int day, int hour, int minute, int second, int microsecond, object tz):
-    return PyDateTimeAPI.DateTime_FromDateAndTime(year, month, day, hour, minute, second, microsecond, tz, PyDateTimeAPI.DateTimeType)
+# Create a time object with a fold using DateTime CAPI factory function.
+cdef inline object time_and_fold_new(int hour, int minute, int second, int microsecond, object tz, int fold):
+    return PyDateTimeAPI.Time_FromTimeAndFold(hour, minute, second, microsecond, tz, fold, PyDateTimeAPI.TimeType)
 
 # Create timedelta object using DateTime CAPI factory function.
 # Note, there are no range checks for any of the arguments.
 cdef inline object timedelta_new(int days, int seconds, int useconds):
     return PyDateTimeAPI.Delta_FromDelta(days, seconds, useconds, 1, PyDateTimeAPI.DeltaType)
 
+# Create a timezone from an offset, optionally a name.
+cdef inline object timezone_new(PyObject* offset, PyObject* name=NULL):
+    return PyDateTimeAPI.TimeZone_FromTimeZone(offset, name)
+
+# Create a datetime object using DateTime CAPI classmethod.
+cdef inline object datetime_from_timestamp(int timestamp, object tz):
+    return PyDateTimeAPI.DateTime_FromTimestamp(PyDateTimeAPI.DateTimeType, (timestamp, tz) if tz is not None else (timestamp,), NULL)
+
+# Create a date object using DateTime CAPI classmethod.
+cdef inline object date_from_timestamp(int timestamp):
+    return PyDateTimeAPI.Date_FromTimestamp(PyDateTimeAPI.DateType, (timestamp,))
+
+
 # More recognizable getters for date/time/datetime/timedelta.
-# There are no setters because datetime.h hasn't them.
-# This is because of immutable nature of these objects by design.
-# If you would change time/date/datetime/timedelta object you need to recreate.
+# There are no setters because datetime.h doesn't have them.
+# This is because of the immutable nature of these objects by design.
+# If you would like to change a time/date/datetime/timedelta object you need to recreate it.
+
+# Get UTC singleton
+cdef inline object get_utc():
+    return <object>PyDateTime_TimeZone_UTC
```

Comment:
This will crash in Py3.6 and earlier, given your `#define` above.
Either there's a way to emulate it in earlier versions, or raise an exception.

Reply:
```suggestion
    if PY_VERSION_HEX < 0x030700b1:
        # TODO: emulate
        raise RuntimeError("Time zones are not available from the C-API")
    return PyDateTime_TimeZone_UTC
```